### DISCUSSION TITLE
Central office > Alano Club

### DISCUSSION THREAD

**Discussion by dougie-wougie** [code4recovery/12-step-meeting-list#1192] [Q&A]
I have all meetings for our district in the Meeting Guide. Awesome. I would like an Alano club to have their meeting in an additional meeting guide on the Alano club site—they receive a read only feed showing only that club's meetings. 

I saw the Import & Export > Import Data Sources, but not sure if that is something I can use. 

The issue for me is I am updating meetings in two places; I prefer one place and feed the other. 

I hope that makes sense.

D

**Comment by lovethecodelabs** [comment]
Short code limiting the meeting output to place id could work,  but in another feed

**Reply by dougie-wougie** [reply]
I am assuming you know how to do that. I am thinking that I need to add another pairing after meetings, but not sure what. e.g., ID=123. Where is a listing of shortcodes? 

https://aascv.org/wp-admin/admin-ajax.php?action=meetings

**Comment by kiyote33** [comment]
Hi @dougie-wougie, 
I think I understand your question. You have two websites: a district site using our 12 Step Meeting List plugin, and an Alano Club site where you would like to display in it's own 12 step meeting list a read only feed of Alano Club meetings from the district site.  This is a common scenario for many sites using our plugin where the first site is where the records are maintained, and the second site which simply displays a filtered view of the records maintained in the first site. Is this what you are after?

**Reply by dougie-wougie** [reply]
That is precisely it. You stated it better. 

Thanks

**Comment by kiyote33** [comment]
One way to accomplish this on the Alano club site is to Import the Data feed from the district site, and filter that feed in a child theme function file. Are you familiar with using a child theme? If not, it's quite easy to setup using the child configurator plugin. I can post the filtering code here if you want to try it out?

**Reply by dougie-wougie** [reply]
I am familiar with the child theme and function file. If you post the filtering code. I will do it. 

d

**Comment by kiyote33** [comment]
Ok, in your alano club child theme functions.php file, start with code to intercept the district meetings feed when importing:
if (!function_exists('tsml_import_reformat')) {
    function tsml_import_reformat($imported_data) {
        //your filtering code goes here
        $meetings = [];
        foreach ($imported_data as $meeting) { 
		if (!$meeting['location'] === 'Alano Club' ) {
                      continue;
		}  else {     
                    $meetings[] = $meeting;
               } 
        }
        return $meetings;
    }
}
The idea here is to filter on the location field for the Alano Club. You could filter on address or another field and get a similar result.  Hope this helps. Let me know if it works for you or if I can be of further assistance.

**Reply by dougie-wougie** [reply]
Thank you. I do not have it working, but working thru it. I assume "location" is the field I see when I export the meetings, not the "location / group" on the UX on the site. I assume I can use any field in the export listing.

D

**Reply by kiyote33** [reply]
Yep. You would see something like **"location":"Alano Club"** in the feed being imported.

**Reply by dougie-wougie** [reply]
OK. I have it working, but not clear the PHP code works on importing a meeting list, not filtering the entire list that is a feed. If so, updates on the district. I need to Export on the district, and import on the Alano site. Do this after each change cycle. 

d

**Reply by kiyote33** [reply]
I'm assuming you've setup the Alano Club site to do a JSON feed Import from the District site. When the importer code checks to see if you have a function named tsml_import_reformat, it runs your filtering code which should drop everything but the Alano Club records.  If you have setup the Change Notification email address on the Settings page, then every night when the Change Detection runs it will scan your disctrict site for changes. When changes are found, the Change Notification email recipients are sent a report which advises them to sign into their site and press the Refresh button on the Data Source import page to get the latest updates from the district site.
Hope you follow all that...

**Reply by kiyote33** [reply]
I'm not sure how that will work with the filtering. I'm thinking it might not. In which case I may have led you in the wrong direction. You will probably have to do what you suggested, simply do the Refresh after a change cycle and leave the Change Detection turned off or ignore the email.
